game-core/src/main/java/games/strategy/triplea/delegate/battle/MustFightBattle.java

```diff
@@ -1494,45 +1496,53 @@ public void execute(final ExecutionStack stack, final IDelegateBridge bridge) {
       }
     };
 
-    final ReturnFire returnFireAgainstAttackingSubs =
-        SubsChecks.returnFireAgainstAttackingSubs(attackingUnits, defendingUnits, gameData);
-    final ReturnFire returnFireAgainstDefendingSubs =
-        SubsChecks.returnFireAgainstDefendingSubs(attackingUnits, defendingUnits, gameData);
-    if (SubsChecks.defenderSubsFireFirst(attackingUnits, defendingUnits, gameData)) {
+    if (firstStrikeStepOrders.contains(FirstStrikeStepOrder.FIRST_STRIKE_DEFENDER_FIRST_NONE)) {
       steps.add(
           new FirstStrikeDefendersFire() {
             private static final long serialVersionUID = 99992L;
 
             @Override
             public void execute(final ExecutionStack stack, final IDelegateBridge bridge) {
-              firstStrikeDefendersFire(returnFireAgainstDefendingSubs);
+              firstStrikeDefendersFire(ReturnFire.NONE);
             }
           });
     }
-    steps.add(
-        new FirstStrikeAttackersFire() {
-          private static final long serialVersionUID = 99991L;
+    if (firstStrikeStepOrders.contains(FirstStrikeStepOrder.FIRST_STRIKE_OFFENDER_ALL)
```

Comment:
This looks to be getting to be quite a bit of logic for `MustFightBattle` to know about. Have you considered making this check be a method on `FirstStrikeStepOrder`?

EG:
```
FirstStrikeStepOrder.containsFirstStrikeAttackersFire(firstStrikeStepOrders)
```

Similar can be applied elsewhere to further remove logic from `MustFightBattle`.

Perhaps it can be made even cleaner by adding a 'Type' attribute to FirstStrikeStepOrder. EG:


```

public enum FirstStrikeType {
  ATTACKER_FIRE,
  DEFENDERS_FIRE,
  DO_NOT_FIRE,
}


FirstStrikeStepOrder.java
  :
  :

  NOT_APPLICABLE(ReturnFire.ALL, FirstStrikeType.DO_NOT_FIRE
  );

  
  @Getter private final ReturnFire returnFire;
  private final FirstStrikeType firstStrikeType;

  static boolean containsAttackerFire(Collection<FirstStrikeStepOrder> order) {
    return order.stream()
        .anyMatch(o -> o.firstStrikeType == FirstStrikeType.ATTACKER_FIRE);
  }

```

WDYT, could any of that help?